### PR TITLE
HWE kernels, KVM, install_rackd support

### PR DESCRIPTION
**Pull request by lurkingsystemsdude**
This commit adds support for:

selecting a specific HWE kernel train
passing the install_kvm boolean flag to build and register a KVM server (new in MAAS 2.5.x)
passing the install_rackd boolean flag to build a MAAS rack controller (new in 2.5.x/2.6)
The install_rackd functionality is untested as of this moment but it should work just like install_kvm.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -3,22 +3,28 @@
...
+LINTER_VERSION ?= 1.17.1# Version of golangci-lint to use in CI
 
 lint: install_lint
 	$(LINTER_BINARY) $(LINTER_OPTIONS) ./...
 .PHONY: lint
 
 install_lint:
 ifeq (,$(shell command -v $(LINTER_BINARY)))
-	$(GOGET) github.com/golangci/golangci-lint/cmd/golangci-lint
+    ifdef CI
```
Comment (robloxrob) [inline Makefile]: the indenting is slightly off

File: README.md [hunk 2]
```diff
@@ -135,6 +135,17 @@ resource "maas_instance" "maas_single_random_node" {
 }
 ```
 
+### Select ubuntu kernel release (HWE kernel version)
+Useful for choosing a particular kernel release train builds
+```
+resource "maas_instance" "maas_single_random_node" {
+    count = 1
+    distro_series = "bionic"
+    hwe_kernel = "hwe-18.04"
```
Comment (robloxrob) [inline README.md:144]: nice

File: README.md [hunk 3]
```diff
@@ -207,6 +218,27 @@ resource "maas_instance" "maas_single_random_node" {
 }
 ```
 
+### Build kvm server
+
+```
+resource "maas_instance" "maas_single_random_node" {
+    count = 1
+
+    install_kvm = true
```
Comment (robloxrob) [inline README.md:227]: nice

File: README.md [hunk 4]
```diff
@@ -207,6 +218,27 @@ resource "maas_instance" "maas_single_random_node" {
...
+}
+```
+
+### Build rack controller
+
+```
+resource "maas_instance" "maas_single_random_node" {
+    count = 1
+
+    install_rackd = true
```
Comment (robloxrob) [inline README.md:237]: nice

File: internal/provider/instance.go [hunk 5]
```diff
@@ -83,6 +84,11 @@ func (i Instance) FromMachine(m maas.Machine) Instance {
 	return i
 }
 
+// GetMetadata implements the Endpoint interface
+func (i *Instance) GetMetadata() interface{} {
+	return maas.Machine{}
```
Comment (robloxrob) [inline internal/provider/instance.go]: I feel like you want to return `maas.Machine` here

**Review by robloxrob — COMMENTED** [review]
This generally seems good. I believe (I could be wrong) you may have pulled in other stuff into this commit. But please review the comments I added here.

File: Makefile (hunk 1 above)
Comment (lurkingsystemsdude) [reply Makefile]: this change does look like something I might have pulled in from one of the FogOps commits while rebasing, but when you look at the code in context, the indention seems fine to me.

File: internal/provider/instance.go (hunk 5 above)
Comment (lurkingsystemsdude) [reply internal/provider/instance.go]: I think you're right.

**Review by robloxrob — APPROVED** [review]
LGTM. Please merge when you are ready and I can cut a release.

**Pull request merged**